packages/gui/hooks/debounced.ts

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useState } from 'react';
```

Comment:
`1.` I as commented above, we may not need this 
`2.` Duplicated code, if we have more than 1 place needed something. Put it under `package/app` or `package/gui` and then other place use from there.